src/Ansatz/Chain.jl

```diff
@@ -227,6 +228,28 @@ function Base.rand(rng::Random.AbstractRNG, sampler::ChainSampler, ::Type{Open},
     Chain(Operator(), Open(), arrays)
 end
 
+@valsplit 2 Tenet.contract(tn::Chain, query::Symbol, site1::Site, site2::Site; direction::Symbol = :left) =
+    Tenet.contract!(copy(tn), Val(query), site1, site2; direction=direction)
+
+function Tenet.contract!(tn::Chain, ::Val{:between}, site1::Site, site2::Site; direction::Symbol = :left)
```

Comment:
Ahh wait, I misunderstood
```suggestion
@valsplit 2 Tenet.contract!(tn::Chain, query::Symbol, args...; kwargs...) = error("Query ':$query' not defined")

Tenet.contract(tn::Chain, query::Symbol, args...; kwargs...) = contract!(copy(tn), query, args...; kwargs...)

function Tenet.contract!(tn::Chain, ::Val{:between}, site1::Site, site2::Site; direction::Symbol = :left)
```